Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -132,7 +132,15 @@ def extract
       loop do
         rev = rev.nil? ? "HEAD" : "#{rev}~1"
         rev, (path,) = Git.last_revision_commit_of_files(repo, pattern, before_commit: rev)
-        odie "Could not find #{name}! The formula or version may not have existed." if rev.nil?
+        if rev.nil? && source_tap.shallow?
+          odie <<~EOS
+            Could not find #{name} but #{source_tap} is a shallow clone! 
+            Try again after running:
+              git -C "$(#{source_tap.path})" fetch --unshallow
```

Comment:
```suggestion
              git -C "#{source_tap.path}" fetch --unshallow
```

Sorry, I'm an idiot! 😭. Last change, I promise!

Reply:
Noo! Thank you for helping! 😃